applications/GeoMechanicsApplication/custom_constitutive/linear_elastic_law.cpp

```diff
@@ -0,0 +1,202 @@
+        // Get Values to compute the constitutive law:
+        Flags& r_flags = rParameterValues.GetOptions();
+
+        // Previous flags saved
+        const bool flag_strain       = r_flags.Is(ConstitutiveLaw::USE_ELEMENT_PROVIDED_STRAIN);
+        const bool flag_const_tensor = r_flags.Is(ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR);
+        const bool flag_stress       = r_flags.Is(ConstitutiveLaw::COMPUTE_STRESS);
+
+        r_flags.Set(ConstitutiveLaw::USE_ELEMENT_PROVIDED_STRAIN, false);
+        r_flags.Set(ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR, true);
+        r_flags.Set(ConstitutiveLaw::COMPUTE_STRESS, true);
+
+        // We compute the stress
+        CalculateMaterialResponseCauchy(rParameterValues);
+        rValue = rParameterValues.GetStressVector();
+
+        // Previous flags restored
+        r_flags.Set(ConstitutiveLaw::USE_ELEMENT_PROVIDED_STRAIN, flag_strain);
+        r_flags.Set(ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR, flag_const_tensor);
+        r_flags.Set(ConstitutiveLaw::COMPUTE_STRESS, flag_stress);
+    }
+
+    return rValue;
+}
+
+Matrix& GeoLinearElasticLaw::CalculateValue(ConstitutiveLaw::Parameters& rParameterValues,
+                                            const Variable<Matrix>& rThisVariable,
+                                            Matrix& rValue)
+{
+    if (rThisVariable == CONSTITUTIVE_MATRIX || rThisVariable == CONSTITUTIVE_MATRIX_PK2 ||
+        rThisVariable == CONSTITUTIVE_MATRIX_KIRCHHOFF) {
+        CalculateElasticMatrix(rValue, rParameterValues);
+    }
+
+    return rValue;
+}
+
+void GeoLinearElasticLaw::SetValue(const Variable<double>&, const double&, const ProcessInfo&)
+{
+    // Since `ConstitutiveLaw::SetValue` always throws an exception, an override
+    // is required when that is undesired behavior
+}
+
+void GeoLinearElasticLaw::SetValue(const Variable<Vector>&, const Vector&, const ProcessInfo&)
+{
+    // Since `ConstitutiveLaw::SetValue` always throws an exception, an override
+    // is required when that is undesired behavior
+}
+
+int GeoLinearElasticLaw::Check(const Properties& rMaterialProperties,
+                               const GeometryType& rElementGeometry,
+                               const ProcessInfo& rCurrentProcessInfo) const
+{
+    KRATOS_ERROR_IF_NOT(rMaterialProperties.Has(YOUNG_MODULUS))
+        << "YOUNG_MODULUS is not available in material parameters" << std::endl;
+    KRATOS_ERROR_IF(rMaterialProperties[YOUNG_MODULUS] <= 0.0)
+        << "YOUNG_MODULUS is invalid value " << std::endl;
+
+    KRATOS_ERROR_IF_NOT(rMaterialProperties.Has(POISSON_RATIO))
+        << "POISSON_RATIO is not available in material parameters" << std::endl;
+
+    const double& nu = rMaterialProperties[POISSON_RATIO];
+    KRATOS_ERROR_IF((nu > 0.499 && nu < 0.501) || (nu < -0.999 && nu > -1.01))
+        << "POISSON_RATIO has invalid value " << std::endl;
```

Comment:
strange range, I would expect -1.0 <= nu < 0.5 ( give or take equal signs )

Reply:
Good catch! Definitely something that needs to be addressed, but not in this PR. I will add an issue to our product backlog to make sure we don't forget about it.

Reply:
Out of scope here, there now is an issue on the product backlog.